### ISSUE TITLE
[Request] Support Torznab API

### ISSUE THREAD

**Issue by kevincox** [HDInnovations/UNIT3D-Community-Edition#4398]
[Torznab](https://torznab.github.io/spec-1.3-draft/index.html) has become the defacto torrent search API due to being supported by the Servarr suite of management tools. Most often it is implemented by some compatibility layer such as [Jackett](https://github.com/Jackett/Jackett) but by implementing it natively in UNIT3D it can be done more efficently and without requiring every user to setup and manage a conversion proxy.

**Comment by james-lawrence** [comment]
id love to see this. is help needed for the implementation?

**Comment by imkylecat** [comment]
Is there any possibility we can get this? If help is needed on the implementation I could step in

**Comment by Roardom** [comment]
I tried previously, but torznab's spec is so ancient that it doesn't support the majority of category-type-resolution combinations that UNIT3D supports. Which would mean we'd need to use custom categories for everything to adhere to the spec, something like this for the category id and automating the category name generation:

> 100000 + 1 * category_id + 100 * resolution_id  + 10000 * type_id

And dozens of separate custom categories.

Alternatively we could not use categories at all and just use torznab tags. One tag for each category, type and resolution prefixed by `category-`, `type-` or `resolution-`.

Another issue is torznab's requirement of offset and limit, while we can only do links to previous and next pages.

Another is the apikey query parameter instead of the bearer token used in UNIT3D.

At the end of the day, Jackett already does the hard work to map out the unit3d api query string required for each filter and no existing torznab client (afaik) has support for manually specifying custom categories or tags anyways.

**Comment by imkylecat** [comment]
That's true. My main frustration is indexer flags not reflecting my group on the *arrs, like if I have global freeleech.
